src/test/java/org/opensearch/neuralsearch/processor/rerank/MLOpenSearchRerankProcessorIT.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.opensearch.neuralsearch.processor.rerank;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.hc.core5.http.HttpHeaders;
+import org.apache.hc.core5.http.io.entity.EntityUtils;
+import org.apache.hc.core5.http.message.BasicHeader;
+import org.junit.After;
+import org.opensearch.client.Request;
+import org.opensearch.client.Response;
+import org.opensearch.common.xcontent.XContentHelper;
+import org.opensearch.common.xcontent.XContentType;
+import org.opensearch.core.rest.RestStatus;
+import org.opensearch.neuralsearch.BaseNeuralSearchIT;
+
+import com.google.common.collect.ImmutableList;
+
+import lombok.SneakyThrows;
+import lombok.extern.log4j.Log4j2;
+
+@Log4j2
+public class MLOpenSearchRerankProcessorIT extends BaseNeuralSearchIT {
+
+    private final static String PIPELINE_NAME = "rerank-mlos-pipeline";
+    private final static String INDEX_NAME = "rerank-test";
+    private final static String TEXT_REP_1 = "Jacques loves fish. Fish make Jacques happy";
+    private final static String TEXT_REP_2 = "Fish like to eat plankton";
+    private final static String INDEX_CONFIG = "{\"mappings\": {\"properties\": {\"text_representation\": {\"type\": \"text\"}}}}";
+    private String modelId;
+
+    @After
+    @SneakyThrows
+    public void tearDown() {
+        super.tearDown();
+        /* this is required to minimize chance of model not being deployed due to open memory CB,
+         * this happens in case we leave model from previous test case. We use new model for every test, and old model
+         * can be undeployed and deleted to free resources after each test case execution.
+         */
+        deleteModel(modelId);
+        deleteSearchPipeline(PIPELINE_NAME);
+        deleteIndex(INDEX_NAME);
+    }
+
+    @SneakyThrows
+    public void testCrossEncoderRerankProcessor() {
+        modelId = uploadTextSimilarityModel();
+        loadModel(modelId);
+        createSearchPipelineViaConfig(modelId, PIPELINE_NAME, "processor/RerankMLOpenSearchPipelineConfiguration.json");
```

Comment:
looks like first 3 methods a better fit for a method annotated with `@Before`

Reply:
ok. I'm gonna keep the pipeline config in the test section though - one day other integration tests may use slightly different configurations.